addons/survey/controllers/survey_session_manage.py

```diff
@@ -178,26 +174,39 @@ def survey_session_code(self, **post):
     @http.route('/s/<string:session_code>', type='http', auth='public', website=True)
     def survey_start_short(self, session_code):
         """" Redirects to 'survey_start' route using a shortened link & token.
-        We match the session_code for open surveys.
+        We match the session_code for existing surveys.
         This route is used in survey sessions where we need short links for people to type. """
+        survey_state = self.survey_check_session_code(session_code)
```

Comment:
Not a good practice to call a json route within another route.
JSON routes are meant to be called using rpc and have a response type meant to be for the frontend only.

Instead, I think we should modify "_fetch_from_session_code" to return not just the survey but also potential errors based on the survey state.
Should be a bit cleaner.

Reply:
Not sure if you are done yet with this one but I was suggestion to have only "_fetch_from_session_code" and merge the result with your "_check_session_error".

So the fetch method will return the survey AND the potential error codes (in a dict).
It should save some code as you usages do "fetch" and then immediately "check" anyway.

Reply:
Sorry, I confused 2 methods, and thought that "_fetch_from_session_code" was called somewhere else without the check but it is not the case, so I will change this